assets/prototype/domain/eventEditor/datetimes/dateCard/EditDateModal.tsx

```diff
@@ -1,20 +1,21 @@
-import { useContext } from 'react';
+import React, { useContext } from 'react';
 import FormModal from '../../../../application/ui/components/forms/FormModal';
 import DatetimeForm from '../dateForm/DateForm';
 import { DateTimeContext } from '../../context/DateTimeProvider';
-import { useEntityMutator, EntityType } from '../../../../application/services/apollo/mutations';
+import { useEntityMutator, EntityType, MutationResult } from '../../../../application/services/apollo/mutations';
 import useTickets from '../../data/queries/tickets/useTickets';
+import { EditDateModalProps } from '../types';
 
-const EditDateModal = ({ relatedTickets, id }) => {
+const EditDateModal: React.FC<EditDateModalProps> = ({ relatedTickets, id }): JSX.Element => {
 	const { editors, editorState } = useContext(DateTimeContext);
 	const isOpen = editorState.isEditorOpen(editors.editForm);
-	const onClose = () => editorState.closeEditor(editors.editForm);
+	const onClose = (): void => editorState.closeEditor(editors.editForm);
 
 	const { updateEntity } = useEntityMutator(EntityType.Datetime, id);
-	const onSubmit = (fields) => updateEntity(fields);
+	const onSubmit = (fields: any): MutationResult => updateEntity(fields);
 	const tickets = useTickets();
 
-	const formComponent = (props) => (
+	const formComponent: React.FC<any> = (props): JSX.Element => (
```

Comment:
is there a built-in interface for props in React?

if not maybe we should create on like:

```ts
interface componentProps {
	[ key: string ]: any
}
```

just to denote that the props at least have to be an object.

thoughts?